const.go

```diff
@@ -0,0 +1,122 @@
+package swag
+
+import (
+	"go/ast"
+	"go/token"
+	"strconv"
+)
+
+// ConstVariable a model to record an const variable
+type ConstVariable struct {
+	Name    *ast.Ident
+	Type    ast.Expr
+	Value   interface{}
+	Comment *ast.CommentGroup
+}
+
+// EvaluateValue evaluate the value
+func (cv *ConstVariable) EvaluateValue(constTable map[string]*ConstVariable) interface{} {
+	if expr, ok := cv.Value.(ast.Expr); ok {
+		value, evalType := evaluateConstValue(cv.Name.Name, cv.Name.Obj.Data.(int), expr, constTable, make(map[string]struct{}))
+		if cv.Type == nil && evalType != nil {
+			cv.Type = evalType
+		}
+		if value != nil {
+			cv.Value = value
+		}
+		return value
+	}
+	return cv.Value
+}
```

Comment:
The behavior of this function is quite unusual. it value is an ast.Expresion , it populates the ConstVariable type and value , otherwise, it returns a value that may be ni ?

Reply:
cv.Value is initialized in the func `collectConstVariables` with a ast.Expr, and then will be reassigned with the evaluated non-nil value of integer or string type，so it would never be nil.